pkg/volume/flexvolume/probe.go

```diff
@@ -127,46 +157,93 @@ func (prober *flexVolumeProber) handleWatchEvent(event fsnotify.Event) error {
 	if err != nil {
 		return err
 	}
-
+	parentPathAbs := filepath.Dir(eventPathAbs)
 	pluginDirAbs, err := filepath.Abs(prober.pluginDir)
 	if err != nil {
 		return err
 	}
 
-	// If the Flexvolume plugin directory is removed, need to recreate it
-	// in order to keep it under watch.
-	if eventOpIs(event, fsnotify.Remove) && eventPathAbs == pluginDirAbs {
-		if err := prober.createPluginDir(); err != nil {
-			return err
+	// event of pluginDirAbs
+	if eventPathAbs == pluginDirAbs {
+		// If the Flexvolume plugin directory is removed, need to recreate it
+		// in order to keep it under watch.
+		if eventOpIs(event, fsnotify.Remove) {
+			if err := prober.createPluginDir(); err != nil {
+				return err
+			}
+			if err := prober.addWatchRecursive(pluginDirAbs); err != nil {
+				return err
+			}
 		}
-		if err := prober.addWatchRecursive(pluginDirAbs); err != nil {
-			return err
+		return nil
+	}
+
+	// event of subdir of pluginDirAbs
+	if parentPathAbs == pluginDirAbs {
```

Comment:
If a driver directory is created under pluginDir After  prober initialization, I think we should add a watch, according to the original code: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/flexvolume/probe.go#L146

Reply:
But it already does without this check: the check at line 193 (`if eventOpIs(event, fsnotify.Create)`) applies for a newly created driver directory, since the pluginDir is always under watch.

Reply:
oh ,got it !